airflow/serialization/serialized_objects.py

```diff
@@ -554,13 +554,13 @@ def deserialize(cls, encoded_var: Any, use_pydantic_models=False) -> Any:
             return SimpleTaskInstance(**cls.deserialize(var))
         elif use_pydantic_models and _ENABLE_AIP_44:
             if type_ == DAT.BASE_JOB:
-                return BaseJobPydantic.parse_obj(var)
+                return JobPydantic.parse_obj(var)
             elif type_ == DAT.TASK_INSTANCE:
                 return TaskInstancePydantic.parse_obj(var)
             elif type_ == DAT.DAG_RUN:
                 return DagRunPydantic.parse_obj(var)
             elif type_ == DAT.DATA_SET:
-                return DatasetPydantic.parse_obj(var)
+                return DatasetPydantic.parse_obj(var) you 
```

Comment:
```suggestion
                return DatasetPydantic.parse_obj(var)
```

Reply:
Yeah. Resolving conflicts from mobile is not the best idea.